src/utils/random.cpp

```diff
@@ -99,42 +99,52 @@ fast_rng_gen& get_fast_io_generator() {
   return ::fast_io_generator;
 }
 
-bool save_rng_to_checkpoint(persist& p, const lbann_comm* comm) {
+bool save_rng_to_checkpoint(persist& p, lbann_comm* comm) {
```

Comment:
Why is `comm` no longer pointer-to-`const`? Why does saving to checkpoint modify `comm`?

Reply:
comm object internally maintains a counter of each call. I am using it here for barrier to make sure the creation of the directory is done only once by the trainer master and visible to everyone else.

Reply:
🤦‍♂.

Counters like this should be marked `mutable`... There is no logical universe in which calling collectives should alter the state of a communicator. Anyway, this isn't your fault (presumably). I'll open an issue.